main.tf

```diff
@@ -13,9 +13,14 @@ locals {
     CreatedOn  = time_static.current.rfc3339
   }, var.tags)
   admin_security_principle_id = var.admin_security_principle_id != null ? var.admin_security_principle_id : data.azurerm_client_config.current.object_id
+  resource_group_id           = var.resource_group_name != null ? data.azurerm_resource_group.existing_graphdb_rg[0].id : azurerm_resource_group.graphdb[0].id
```

Comment:
You can define multiple `locals {}` sections. So my suggestion here would be to define

```
locals {
  resource_group_id    = ...
  resource_group_name  = ....
}
```

after `resource "azurerm_resource_group" "graphdb" {` & `data "azurerm_resource_group" "existing_graphdb_rg" {`

This way would follow the order of definitions.